UHResidentInformationAPI/V1/Gateways/UHGateway.cs

```diff
@@ -21,7 +24,52 @@ public List<ResidentInformation> GetAllResidents(string houseReference, string r
 
         public ResidentInformation GetResidentById(string houseReference, int personReference)
         {
-            return null;
+            var databaseRecord = _uHContext.Persons.FirstOrDefault(p => p.HouseRef == houseReference && p.PersonNo.Equals(personReference));
+
+            if (databaseRecord == null) return null;
+
+            var addressForPerson =
+                _uHContext.Addresses.OrderByDescending(a => a.Dtstamp).FirstOrDefault(a => a.HouseRef == databaseRecord.HouseRef);
+
+            var contactLinkNoForPerson = GetContactNoFromContactLink(databaseRecord.HouseRef);
+
+            var telephoneNumberForPerson = _uHContext.TelephoneNumbers.Where(t => t.ContactID == contactLinkNoForPerson).ToList();
+
+            var emailAddressForPerson =
+                _uHContext.EmailAddresses.Where(c => c.ContactID == contactLinkNoForPerson).ToList();
+
+            var person = MapPersonAndAddressesToResidentInformation(databaseRecord, addressForPerson);
+
+            AttachContactDetailsToPerson(person, telephoneNumberForPerson, emailAddressForPerson);
+
+            return person;
         }
+
+        private static ResidentInformation MapPersonAndAddressesToResidentInformation(Person person, Address address)
+        {
+            var resident = person.ToDomain();
+
+            if (address == null) return resident;
+
+            resident.Address = address.ToDomain();
+            resident.UPRN = address.UPRN;
+
+            return resident;
+        }
+
+        private static void AttachContactDetailsToPerson(ResidentInformation person, List<TelephoneNumber> phoneNumbers,
+            List<EmailAddresses> emails)
+        {
+            person.PhoneNumber = phoneNumbers.Any() ? phoneNumbers.Select(n => n.ToDomain()).ToList() : null;
+            person.Email = emails.Any() ? emails.Select(n => n.ToDomain()).ToList() : null;
+        }
+
+        private int? GetContactNoFromContactLink(string houseReference)
+        {
+            var tagReference = _uHContext.TenancyAgreements.FirstOrDefault(ta => ta.HouseRef == houseReference)?.TagRef;
+            var contactLinkUsingTagReference = _uHContext.ContactLinks.FirstOrDefault(co => co.TagRef == tagReference)?.ContactID;
```

Comment:
This might return more than one ContactNo in case of joint tenancy. I know you have the first or default, but if we want the contact details for a specific person within a household, you need to also query by “Key2”, which is a column referring to the “person_no” column in the members table.

Reply:
Changes added in ca97063